ksql-serde/src/main/java/io/confluent/ksql/serde/json/KsqlJsonDeserializer.java

```diff
@@ -132,12 +132,18 @@ private Object enforceFieldType(Schema fieldSchema, JsonNode fieldJsonNode) {
       Iterator<String> fieldNames = jsonNode.fieldNames();
       while (fieldNames.hasNext()) {
         String fieldName = fieldNames.next();
-        keyMap.put(fieldName.toUpperCase(), fieldName);
+        if (fieldName.startsWith("@")) {
```

Comment:
Is it possible that the fieldName could just be `@`? in which case the next line would fail

Reply:
Field name cannot be '@' and if we have '@' as field name we should fail anyway.

Reply:
Maybe we should fail with a more meaningful exception? i.e.., not `IndexOutOfBoundsException`

Reply:
Added better error message for field name '@'.